### PR TITLE
Conversion Service is no longer generic

### PR DESCRIPTION
**Pull request by timyates**
Caffeine cache is still failing as the configuration is not being picked up

### CODE REVIEW

File: cache-core/src/main/java/io/micronaut/cache/CacheConfiguration.java [hunk 1]
```diff
@@ -81,22 +79,22 @@ public String getCacheName() {
     /**
      * @return The initial capacity of the cache
      */
-    public OptionalInt getInitialCapacity() {
-        return initialCapacity == null ? OptionalInt.empty() : OptionalInt.of(initialCapacity);
+    public Optional<Integer> getInitialCapacity() {
+        return initialCapacity == null ? Optional.empty() : Optional.of(initialCapacity);
     }
```
[the remark this answers is not in the document]
Comment (sdelamo) [reply cache-core/src/main/java/io/micronaut/cache/CacheConfiguration.java:84]: @timyates can you create an issue in core explaining what was possible before what it is not possible now.

**Review by sdelamo — APPROVED** [review]

**Comment by timyates** [comment]
Waiting for https://github.com/micronaut-projects/micronaut-core/pull/8269 to be published to a SNAPSHOT, and I'll test reverting the Optional changes 👍

File: cache-core/src/main/java/io/micronaut/cache/CacheConfiguration.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (sdelamo) [reply cache-core/src/main/java/io/micronaut/cache/CacheConfiguration.java:84]: @dstepanov fixed here https://github.com/micronaut-projects/micronaut-core/pull/8269

[the remark this answers is not in the document]
Comment (timyates) [reply cache-core/src/main/java/io/micronaut/cache/CacheConfiguration.java:84]: https://github.com/micronaut-projects/micronaut-cache/pull/521#issuecomment-1299943110

**Comment by timyates** [comment]
@sdelamo OptionalLong and OptionalInt back in and works 🎉

**Pull request merged**